defopt.py

```diff
@@ -218,14 +219,33 @@ def _populate_parser(func, parser, parsers, short):
             if param.kind == param.VAR_POSITIONAL:
                 kwargs['action'] = 'append'
                 kwargs['default'] = []
-        if inspect.isclass(type_.type) and issubclass(type_.type, Enum):
+        make_tuple = member_types = None
+        if _is_generic_type(type_.type, Tuple):
+            make_tuple = tuple
+            member_types = type_.type.__args__
+            kwargs['nargs'] = len(member_types)
+        elif (inspect.isclass(type_.type) and issubclass(type_.type, tuple)
+              and hasattr(type_.type, '_fields')
+              and hasattr(type_.type, '_field_types')):
+            # On Py2, `_field_types` does not preserve order, so retrieve the
+            # order from `_fields`.
+            make_tuple = lambda args: type_.type(*args)
+            member_types = tuple(type_.type._field_types[field]
+                                 for field in type_.type._fields)
+            kwargs['nargs'] = len(member_types)
+            if not positional:  # http://bugs.python.org/issue14074
+                kwargs['metavar'] = type_.type._fields
+        elif inspect.isclass(type_.type) and issubclass(type_.type, Enum):
             # Want these to behave like argparse choices.
             kwargs['choices'] = _ValueOrderedDict(
                 (x.name, x) for x in type_.type)
             kwargs['type'] = _enum_getter(type_.type)
         else:
             kwargs['type'] = _get_parser(type_.type, parsers)
-        _add_argument(parser, name, short, **kwargs)
+        action = _add_argument(parser, name, short, **kwargs)
+        if make_tuple:
+            action.__class__ = _make_store_tuple_action(
+                make_tuple, member_types, parsers)
```

Comment:
The `action` kwarg to `add_argument` can be an Action subclass. It would be cleaner to pass that through rather than modifying the action after the fact.